src/gtfs/read.rs

```diff
@@ -879,6 +881,155 @@ pub fn set_dataset_validity_period(
     Ok(())
 }
 
+#[derivative(Default)]
+#[derive(Derivative, Deserialize, Debug, Clone, PartialEq)]
+enum FrequencyExactTime {
+    #[derivative(Default)]
+    #[serde(rename = "0")]
+    Inexact,
+    #[serde(rename = "1")]
+    Exact,
+}
+
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+struct Frequency {
+    trip_id: String,
+    start_time: Time,
+    end_time: Time,
+    headway_secs: u32,
+    #[serde(default, deserialize_with = "de_with_empty_default")]
+    exact_times: FrequencyExactTime,
+}
+
+pub fn manage_frequencies<P: AsRef<path::Path>>(
+    collections: &mut Collections,
+    path: P,
+) -> Result<()> {
+    info!("Reading frequencies.txt");
+    let path = path.as_ref();
+    let frequencies_path = path.join("frequencies.txt");
+    let mut rdr =
+        csv::Reader::from_path(&frequencies_path).with_context(ctx_from_path!(frequencies_path))?;
+    let gtfs_frequencies: Vec<Frequency> = rdr
+        .deserialize()
+        .collect::<StdResult<_, _>>()
+        .with_context(ctx_from_path!(frequencies_path))?;
+    let mut trip_id_sequence: HashMap<String, u8> = HashMap::new();
+    let mut vehicle_journeys = collections.vehicle_journeys.take();
+    for frequency in &gtfs_frequencies {
+        if frequency.start_time == frequency.end_time {
+            warn!(
+                "frequency for trip {:?} has same start and end time",
+                frequency.trip_id
+            );
+            continue;
+        }
+        let datetime_estimated = match frequency.exact_times {
+            FrequencyExactTime::Exact => false,
+            FrequencyExactTime::Inexact => true,
+        };
+        let corresponding_vj = skip_fail!(
+            vehicle_journeys
+                .iter()
+                .find(|&vj| vj.id == frequency.trip_id)
+                .cloned()
+                .ok_or_else(|| format_err!(
+                    "frequency mapped to an unexisting trip {:?}",
+                    frequency.trip_id
+                ))
+        );
+        let mut codes = corresponding_vj.codes.clone();
+        codes.insert(("source".to_string(), frequency.trip_id.clone()));
+        let mut start_time = frequency.start_time;
+        if corresponding_vj.stop_times.is_empty() {
+            warn!(
+                "frequency mapped to trip {:?} with no stop_times",
+                frequency.trip_id
+            );
+            continue;
+        }
+        let mut arrival_time_delta = corresponding_vj
+            .stop_times
+            .iter()
+            .min()
+            .unwrap()
+            .arrival_time;
```

Comment:
You can mutualize these 2 lines to avoid unwrap:
```rust
let mut arrival_time_delta = match corresponding_vj.stop_times.iter().min() {
    None => {
      warn!(
                "frequency mapped to trip {:?} with no stop_times",
                frequency.trip_id
            );
            continue;
    }
    Some(st) => st.arrival_time,
};
```

Reply:
:+1: